### PR TITLE
[1848] output BoE revenue history

### PR DESCRIPTION
**Pull request by benjaminxscott**
### Before clicking "Create"

- [x] Branch is derived from the latest `master`
- [NA] Add the `pins` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

### Implementation Notes

* **Explanation of Change**

Output revenue history for BoE

* **Screenshots**

before:
Screenshot 2023-11-11 5 20 28 PM

after:
Screenshot 2023-11-11 5 05 11 PM

* **Any Assumptions / Hacks**

### CODE REVIEW

File: lib/engine/game/g_1848/step/dividend.rb [hunk 1]
```diff
@@ -38,6 +38,13 @@ def process_dividend(action)
             revenue = calculate_boe_revenue
             payout = send(:payout, entity, revenue)
             payout_shares(entity, revenue) if payout[:per_share].positive?
+            action.id = @game.actions.last.id
```
Comment (bentziaxl) [inline lib/engine/game/g_1848/step/dividend.rb:41]: why is this line needed?

Comment (benjaminxscott) [reply lib/engine/game/g_1848/step/dividend.rb:41]: `action` needs an 'id' or else it won't render BoE's revenue as a clickable hyperlink when stored in the operating history

The BoE revenue isn't a "real action" that's stored in the state, so we just set it to the last real action. 

I do the same thing here for the case when a company skips its revenue step on its first operating round
https://github.com/benjaminxscott/18xx/blame/57f9f3e26577523aa7ce77b3a2850c8f8f668adf/lib/engine/step/dividend.rb#L36
